connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java

```diff
@@ -381,6 +435,10 @@ Map<TopicPartition, Long> readEndOffsets(Set<TopicPartition> assignment, boolean
             } catch (UnsupportedVersionException e) {
                 // This may happen with really old brokers that don't support the auto topic creation
                 // field in metadata requests
+                if (requireAdminForOffsets) {
+                    // Should be handled by the caller during log startup
+                    throw e;
```

Comment:
Should we set `admin = null` in this case, like the comment said:
`// Forget the reference to the admin so that we won't even try to use the admin the next time this method is called` ?

Reply:
We want to keep throwing the exception here if the method is called again; nullifying the `admin` field would stop that from happening. That does remind me though, we should probably copy the same error message from [here](https://github.com/apache/kafka/blob/f421c008aa0495441ce6f017cdd041937a26d10f/connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java#L333-L338) into the code path for starting connector offset stores. I've done that in my latest push; LMKWYT.

Reply:
SGTM. Thanks.